### DISCUSSION TITLE
should have a stable version and edge version of this extension

### DISCUSSION THREAD

**Discussion by tigerinus** [redhat-developer/vscode-java#2082] [Ideas]
We heavily depend on this extension for our day to day Java development. Thanks vscode-java team for all the hard work!

However, recent updates of this extension have caused different issues here and there.

Can we adopt the model that there is a versioning on the slow ring, aka `stable`; and a versioning on the fast ring, aka `edge`?

This way the team won't be interrupted by installing the `stable` version. 

Thanks!

**Comment by nickzhums** [comment]
Hi @tigerinus this is Nick and I'm PM for VS Code Java. Would love to learn more about your use cases and scenarios, would you mind sharing more details on your Java development? If you don't mind, we can contact you via email ?

**Comment by CsCherrYY** [comment]
Currently there is a discussion about insider extensions in VSCode repo. In the discussion:
- In the long term (18 months in their plan), VSCode marketplace will add support for pre-release versions, they are `edge` versions we're discussing here IMO. 
- In the short term (October for VSCode insiders in their milestone), they have a proposal about `--insiders` flag to support two versioning channels, see https://github.com/microsoft/vscode/issues/15756#issuecomment-916728348. 

The proposal is still under discussing. So we can post some comments from Java side there.

**Reply by rgrunber** [reply]
As of [vscode-java 1.7.0](https://github.com/redhat-developer/vscode-java/issues/2285), we support publishing pre-releases. They are .vsix derived from the commits on our development branch, and a new one is published daily (if there's new content). One just needs to enable pre-releases for the vscode-java extension to subscribe to this stream.
